servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/gc/AbstractRestGCTest.java

```diff
@@ -0,0 +1,370 @@
+            branch3.getHash(),
+            46,
+            "cid_1",
+            "table_1",
+            "file5",
+            b2table1.content,
+            null);
+    // delete branch3 should not affect as it is performed after cutoff timestamp.
+    deleteBranch(branch3.getName(), b3table1.hash);
+
+    // drop table "table_1" on branch1 should not affect as it is performed after cutoff timestamp.
+    dropTableCommit(prefix, branch1, b1table1.hash, "table_1");
+
+    performGc(cutoffTime, null, expectedExpired, true);
+  }
+
+  @Test
+  public void testMultiRefCutoffTimeStampPerRef() throws BaseNessieClientServerException {
+    String prefix = "multiRefCutoffTimeStampPerRef";
+    Map<String, Map<String, Set<Content>>> expectedExpired = new HashMap<>();
+
+    Branch branch1 = createBranch(prefix + "_1");
+    Branch branch2 = createBranch(prefix + "_2");
+    // commit for "table_1" on branch1
+    CommitOutput table1 =
+        commitSingleOp(
+            prefix, branch1, branch1.getHash(), 42, "cid_1", "table_1", "file1", null, null);
+    // expect this commit to be expired as it is before cutoff time.
+    fillExpectedContents(branch1, 1, expectedExpired);
+
+    // commit for "table_2" on branch2
+    CommitOutput table2 =
+        commitSingleOp(
+            prefix, branch2, branch2.getHash(), 42, "cid_2", "table_2", "file1", null, null);
+    // expect this commit to be expired as it is before cutoff time.
+    fillExpectedContents(branch2, 1, expectedExpired);
+
+    Instant defaultCutoffTime = Instant.now();
+
+    // commits for "table_1"
+    table1 =
+        commitSingleOp(
+            prefix, branch1, table1.hash, 44, "cid_1", "table_1", "file3", table1.content, null);
+    table1 =
+        commitSingleOp(
+            prefix, branch1, table1.hash, 45, "cid_1", "table_1", "file4", table1.content, null);
+
+    // commits for "table_2"
+    table2 =
+        commitSingleOp(
+            prefix, branch2, table2.hash, 44, "cid_2", "table_2", "file3", table2.content, null);
+    // expect this commit to be expired as it is before cutoff time for branch2's per ref cutoff
+    // time and branch2's table_2 has some more commits down.
+    fillExpectedContents(branch2, 1, expectedExpired);
+
+    Instant branch2CutoffTime = Instant.now();
+    Map<String, Instant> perRefCutoffTime = new HashMap<>();
+    perRefCutoffTime.put(branch2.getName(), branch2CutoffTime);
+
+    // commits for "table_2"
+    table2 =
+        commitSingleOp(
+            prefix, branch2, table2.hash, 45, "cid_2", "table_2", "file4", table2.content, null);
+
+    performGc(defaultCutoffTime, perRefCutoffTime, expectedExpired, true);
+  }
+
+  @Test
+  public void testMultiRefAssignRef() throws BaseNessieClientServerException {
+    String prefix = "multiRefAssignRef";
+    Map<String, Map<String, Set<Content>>> expectedExpired = new HashMap<>();
+
+    Branch before = createBranch(prefix + "_0");
+    Branch branch1 = createBranch(prefix + "_1");
```

Comment:
Lines 333-359 and 162-187 are identical - so are the test cases actually. So worth to either parameterize or extract the common parts and move those tests next to each other.

Reply:
merged the test cases as these are after cutoff time operations and doesn't really impact.